provider/aws/aws.go

```diff
@@ -993,6 +995,28 @@ func changesByZone(zones map[string]*route53.HostedZone, changeSet Route53Change
 			continue
 		}
 		for _, z := range zones {
+			// IMPORTANT EXPLAIN: I tried to fix this in here but a lot of tests started to fail in cascade.
+			//                    Found that nil recordTypes are entering in this function
```

Comment:
When are "nil recordTypes" entering this loop?

Reply:
> When are "nil recordTypes" entering this loop?

func TestAWSChangesByZones(t *testing.T) { in aws_test.go is running tests providing empty RecordTypes on the function